analyzers/src/SonarAnalyzer.Common/Helpers/KnownReference.Predicates.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers;
+
+public sealed partial class KnownReference
+{
+    internal static Func<AssemblyIdentity, bool> NameIs(string name) =>
+        x => x.Name.Equals(name);
+
+    internal static Func<AssemblyIdentity, bool> StartsWith(string name) =>
+        x => x.Name.StartsWith(name);
+
+    internal static Func<AssemblyIdentity, bool> EndsWith(string name) =>
+        x => x.Name.EndsWith(name);
+
+    internal static Func<AssemblyIdentity, bool> Contains(string name) =>
+        x => x.Name.Contains(name);
+
+    internal static Func<AssemblyIdentity, bool> VersionLowerThen(string version) =>
+        VersionLowerThen(Version.Parse(version));
+
+    internal static Func<AssemblyIdentity, bool> VersionLowerThen(Version version) =>
+        x => x.Version < version;
+
+    internal static Func<AssemblyIdentity, bool> VersionGreaterOrEqual(string version) =>
+        VersionGreaterOrEqual(Version.Parse(version));
+
+    internal static Func<AssemblyIdentity, bool> VersionGreaterOrEqual(Version version) =>
+        x => x.Version >= version;
+
+    internal static Func<AssemblyIdentity, bool> VersionBetween(string from, string to) =>
+        VersionBetween(Version.Parse(from), Version.Parse(to));
+
+    internal static Func<AssemblyIdentity, bool> VersionBetween(Version from, Version to) =>
+        x => x.Version >= from && x.Version <= to;
+
+    internal static Func<AssemblyIdentity, bool> OptionalPublicKeyTokenIs(string key) =>
+        x => !x.HasPublicKey || PublicKeyEqualHex(x, key);
+
+    internal static Func<AssemblyIdentity, bool> PublicKeyTokenIs(string key) =>
+        x => x.HasPublicKey && PublicKeyEqualHex(x, key);
```

Comment:
Same as above
```suggestion
```

Reply:
If there is a strong name it is the most reliable source of identity. Also if we don't add the possibility to check for the strong name no one will ever add the functionality even in cases where it would be useful. The functionality is now there and fully covered by tests with our own public key.

Reply:
We don't need to be that precise at all. We'll likely use these for an early bailout, as you plan to use them. And there will be no harm if we trigger on something that is a false match.

Let's remove it.

Reply:
I also highly doubt that we would ever track a library that updates it's public key between versions.